applications/nrf_desktop/src/modules/watchdog.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/watchdog.h>
+
+#define MODULE watchdog
+#include "module_state_event.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_WATCHDOG_LOG_LEVEL);
+
+#define WDT_FEED_WORKER_DELAY_MS ((CONFIG_DESKTOP_WATCHDOG_TIMEOUT)/3)
+
+struct wdt_data_storage {
+	struct device *wdt_drv;
+	int wdt_channel_id;
+	struct k_delayed_work work;
+};
+static struct wdt_data_storage wdt_data;
+
+static void watchdog_feed_worker(struct k_work *work_desc)
+{
+	struct wdt_data_storage *data =
+			CONTAINER_OF(work_desc, struct wdt_data_storage, work);
+
+	int err = wdt_feed(data->wdt_drv, data->wdt_channel_id);
+
+	if (err) {
+		LOG_ERR("Cannot feed watchdog. Error code: %d", err);
+		module_set_state(MODULE_STATE_ERROR);
+	} else {
+		k_delayed_work_submit(&data->work, WDT_FEED_WORKER_DELAY_MS);
+	}
+}
+
+static void wdt_timeout_callback(struct device *dev, int channel_id)
+{
+	ARG_UNUSED(dev);
+	ARG_UNUSED(channel_id);
+	LOG_ERR("Watchdog timeout elapsed. Restarting the device!");
+}
+
+static int watchdog_timeout_install(struct wdt_data_storage *data)
+{
+	static const struct wdt_timeout_cfg wdt_settings = {
+			.window = {
+				.min = 0,
+				.max = CONFIG_DESKTOP_WATCHDOG_TIMEOUT
+			},
+			.callback = wdt_timeout_callback,
```

Comment:
set callback to NULL if possible as we don't do anything interesting here anyway

Reply:
callback removed